src/System.Management.Automation/engine/regex.cs

```diff
@@ -142,7 +142,11 @@ StringComparison GetStringComparison()
                 {
                     stringComparison = Options.HasFlag(WildcardOptions.CultureInvariant)
                         ? StringComparison.InvariantCultureIgnoreCase
-                        : StringComparison.CurrentCultureIgnoreCase;
+                        : CultureInfo.CurrentCulture.Name.Equals("en-US-POSIX", StringComparison.OrdinalIgnoreCase)
```

Comment:
Looking how we use WildcardPattern.IsMatch() I am sure that in most of cases we can definitely replace `StringComparison.CurrentCultureIgnoreCase` with `StringComparison.OrdinalIgnoreCase`. One place where we should be careful is HelpSystem but where we could use Regex directly.

Reply:
I'm not sure about it. Quoted from doc about [StringComparer.OrdinalIgnoreCase](https://docs.microsoft.com/en-us/dotnet/api/system.stringcomparer.ordinalignorecase?view=netframework-4.8#remarks):
>The StringComparer returned by the OrdinalIgnoreCase property treats the characters in the strings to compare as if they were converted to uppercase **using the conventions of the invariant culture**, and then performs a simple byte comparison that is independent of language.

So when using `OrdinalIgnoreCase`, it will actually use `invariant culture` to convert characters to upper cases. This makes me nervous about using `OrdinalIgnoreCase` in general for wildcard matching.

Reply:
Sorry, my comment was out of the context. I meant that we do culture-sensitive comparisons while we should be culture-insensitive (ordinal) in most of cases where WildcardPattern is used.

As for your fix  it is LGTM.